### DISCUSSION TITLE
Scroll panel sizing

### DISCUSSION THREAD

**Discussion by SamSandq** [frang75/nappgui_src#79] [General]
I'm writing a photo management app, and want to show the photos in a scrolling panel, with set width, and flexible height to accommodate from 1 to thousands of photos.

However, the panel_scroll gets its size from

panel_size(collection, s2df(HSIZE, VSIZE));

with e.g. 1000, 800. When wanting to show many photos it decreases the size of each photo to fit into the size, which I do not want -- I want the height of the panel to be whatever size that will fit my photos with a given size (i.e., number of columns fixed, but rows not). It will not allow me to change the VSIZE to anything but the enclosing panel/layout.

Any ideas how to achieve this?

**Comment by frang75** [comment]
Hi @SamSandq!
The Panel object is, in essence, a container of child controls and perhaps is not the best option to implement a photo album. I think the best option is a View control with scrollbars. With that will be free to implement any kind of drawing and mouse effects over the (selected, focused, mouse-over) images. Also adding titles and other info.

You can use this demo app as a basis, just changing the rectangles with images.

https://nappgui.com/en/howto/drawbig.html

Therefore, support for Dynamic layout and dynamic panels is planned for future NAppGUI releases. https://github.com/frang75/nappgui_src/issues/74

**Reply by SamSandq** [reply]
I have now implemented what you suggested, and it's a good and fast solution. Will continue with more functionality, now that the basic mechanism for photo collections is known.

Many thanks for the suggestion!

And will now port it to 1.4, which had some issues...

Happy new year, ¡y un próspero año nuevo a ti!
